File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        //  HashTable 이용 방법
        // unordered_map<int,int> mp;
        // for(int i = 0; i < nums.size(); i++){
        //     if(mp[nums[i]] == 0){
        //         mp[nums[i]]++;
        //     }else{
        //         return true;
        //     }
        // }
        // return false;

        //  정렬 후 원소를 비교하여 같은 것이 있다면 최소 2개 있다는 뜻
        //  true를 반환
        sort(nums.begin(), nums.end());
        for(int i = 1; i < nums.size(); i++){
            if(nums[i] == nums[i-1]){
                return true;
            }
        }
        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.cs
public class Solution {
    public bool ContainsDuplicate(int[] nums) {
        //  https://leetcode.com/problems/contains-duplicate/solutions/1836662/faster-than-97-easy-to-read-one-liner/
        //  nums를 이용해 만든 HashSet의 Count가 nums.Length 보다 작으면 중복되는 원소가 있다는 뜻 => true 반환
        //  같으면 없다는 뜻 => false 반환
        return new HashSet<int>(nums).Count < nums.Length;
    }
}